core/src/collections.rs

```diff
@@ -85,29 +89,38 @@ impl MongoCollections {
         collection_name_filter: &str,
         table_type: &str,
     ) -> Self {
-        let names = mongo_connection
+        let db_name_filter_regex = to_name_regex(db_name_filter);
+        let databases = mongo_connection
             .client
-            .list_database_names(to_name_regex_doc(db_name_filter), None)
-            .unwrap();
-        let mut databases: Vec<CollectionsForDb> = Vec::with_capacity(names.len());
-        for name in names {
-            let list_coll_name_filter = to_name_regex_doc(collection_name_filter);
-            let db = mongo_connection.client.database(name.as_str());
-            let collections = db
-                .list_collections(
-                    add_table_type_filter(table_type, list_coll_name_filter),
-                    None,
-                )
-                .unwrap();
-            databases.push(CollectionsForDb {
-                database_name: name,
-                collection_list: collections,
-            });
-        }
+            .list_database_names(None, None)
+            .unwrap()
+            .iter()
+            .filter(|val| {
+                if db_name_filter_regex.is_some() {
```

Comment:
[nit] since we have `is_some` with an `else` I think this would look better just as a match (fewer braces).

Reply:
Also means removing an `unwrap`!

Reply:
I like that :D